src/routes/news/containers/NewsPage.js

```diff
@@ -0,0 +1,101 @@
+// @flow
+
+import * as React from 'react'
+import { connect } from 'react-redux'
+import compose from 'lodash/fp/compose'
+import { LocalNewsModel } from '@integreat-app/integreat-api-client'
+import ContentNotFoundError from '../../../modules/common/errors/ContentNotFoundError'
+import FailureSwitcher from '../../../modules/common/components/FailureSwitcher'
+import type { TFunction } from 'react-i18next'
+import { withTranslation } from 'react-i18next'
+import type { StateType } from '../../../modules/app/StateType'
+import NewsElement from '../components/NewsElement'
+import NewsList from '../components/NewsList'
+import Tabs from '../components/Tabs'
+import PushNewsDetails from '../components/PushNewsDetails'
+
+type PropsType = {|
+  news: Array<LocalNewsModel>,
+  city: string,
+  newsId: ?string,
+  language: string,
+  t: TFunction,
+  path: string
+|}
+
+const LOCAL_NEWS = 'local'
+const TU_NEWS = 'tu'
+
+/**
+ * Displays a list of news or a single newsElement, matching the route /<location>/<language>/news(/<id>)
+ */
+export class NewsPage extends React.Component<PropsType> {
+  state = {
+    hasNextPage: true,
+    isNextPageLoading: false,
+    items: []
+  };
+
+  renderNewsElement = (language: string, type: string) => (newsItem: EventModel) => (
+    <NewsElement newsItem={newsItem} language={language} key={newsItem.path} type={type} path={this.props.path} />
+  )
+
+  render () {
+    const { news, path, newsId, city, language, t } = this.props
+
+    if (newsId) {
+      const newsItem = news.find(_newsItem => {
+        return _newsItem.title === decodeURIComponent(newsId)
+      })
+
+      if (newsItem) {
+        const { title, message, timestap } = newsItem
+        return (
+          <>
+            <PushNewsDetails title={title} message={message} timestap={timestap} />
+          </>
+        )
+      } else {
+        const error = new ContentNotFoundError({ type: 'newsItem', id: newsId, city, language })
+        return <FailureSwitcher error={error} />
+      }
+    }
+    return (
+      <>
+        <Tabs>
+        {/* TODO: update the locale file realted issues */}
+          <div label={t('LOCAL NEWS')} type={LOCAL_NEWS}>
+            <NewsList
+              noItemsMessage={t('currentlyNoEvents')}
+              isNewsList
+              items={news}
+              renderItem={this.renderNewsElement(language, LOCAL_NEWS)}
```

Comment:
I think its better if we use render prop here rather than this way, what do you think?
https://reactjs.org/docs/render-props.html